core/bootstrap/bootstrap.go

```diff
@@ -125,7 +125,9 @@ func Bootstrap(id peer.ID, host host.Host, rt routing.Routing, cfg BootstrapConf
 	doneWithRound <- struct{}{}
 	close(doneWithRound) // it no longer blocks periodic
 
-	startSavePeersAsTemporaryBootstrapProc(cfg, host, proc)
+	if cfg.LoadBackupBootstrapPeers != nil && cfg.SaveBackupBootstrapPeers != nil {
```

Comment:
This is not very resistant to mistakes, what happen if LoadBackupBootstrapPeers is not set ?
The code will silently not save backups even tho someone could think it does.

I think this should be moved to the [functional option pattern](https://golang.cafe/blog/golang-functional-options-pattern.html) then in the option handler you panic or error if the handler is nil.
Your handler would also take both functions so it's clear that it's not supported to only have a saver.
You maybe could have an option for just a loader but I no one asked for it so let's not make it more complicated.